redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -275,20 +275,26 @@ inline void
                     else if (strcmp(dbusError->name,
                                     "xyz.openbmc_project.Common.Error."
                                     "Unavailable") == 0)
+                    {
+                        messages::serviceDisabled(
+                            aResp->res, "/redfish/v1/Systems/system/"
+                                        "LogServices/HardwareIsolation");
+                    }
+                    else if (strcmp(dbusError->name,
+                                    "xyz.openbmc_project.Common.Error."
+                                    "InsufficientPermission") == 0)
                     {
                         // The Enabled property value will be "true" to
                         // de-isolate.
                         constexpr bool enabledPropVal = true;
                         messages::propertyValueResourceConflict(
```

Comment:
Why is InsufficientPermission mapped to propertyValueResourceConflict ?

Reply:
The hardware isolation application will reject the request with an `InsufficientPermission` error If the user tried to configure/deconfigure the hardware that's isolated the system due to some failure.

Here, using the _PropertyValueResourceConflict_ error for the above case to say the user like `"The property 'Enabled' with the requested value of 'true' could not be written because the value conflicts with the state or configuration of the resource at '/redfish/v1/Systems/system/LogServices/HardwareIsolation/Entries/1'."` so he knows why it is not allowed by looking that at the pointed URI i.e he can know the severity of the deconfigured resource.

Reply:
I think the context is different from the Redfish URI point of view, right?  i.e one place **the User using the property** (i.e The Enabled property as per Processor schema for the Core and Memory schema for the DIMM) and **the User using the HTTP DELETE request** (i.e DELETE request in the HardwareIsolation LogEntry) to deconfigure it.

Please share your view on this context or You are referring to preferring the 1-to-1 error mapping i.e DBus to Redfish even if the user context (aka URI) is different? IMO, we should give a detailed error message for a better understanding rather than sticking with 1-to-1 mapping.

If you have a different opinion, please share, I like to listen to your thoughts.

Reply:
Done, changed to use `ResourceCannotBeDeleted` as pre request.